src/daemon/ca/routes.rs

```diff
@@ -626,24 +617,28 @@ impl Roas {
     /// authorizations change, or if ROAs are about to expire.
     pub fn update(
         &self,
-        routes: &Routes,
+        all_routes: &Routes,
         certified_key: &CertifiedKey,
         config: &Config,
         signer: &KrillSigner,
     ) -> KrillResult<RoaUpdates> {
+        let relevant_routes = all_routes.filter(certified_key.incoming_cert().resources());
```

Comment:
This seems to diff to the previous implementation. I tried following the caller chain back several levels and didn't see the old calling code doing such filtering, maybe it's there if I were to keep looking, or is this actually a change and if so is it deliberate?